### DISCUSSION TITLE
What's the best way to ensure the longevity of this project? Contributing code? Setting up donations?

### DISCUSSION THREAD

**Discussion by davidatsurge** [sycamore-rs/sycamore#448] [General]
Thank you for the work that you put into this library!

I'm very interested in using this library, but I have worries about it's longevity, since it's relatively new, and doesn't have massive adoption yet.

I wanted to ask what would be most helpful in ensuring longevity. If code contributions are the absolute best way to do that, I will try my best to contribute. Would perhaps donations would also help incentivize work on this project? I realize that, at the beginning at least, there will be very little donations coming in. But I think it wouldn't hurt to have Github Sponsors setup so that any interested parties (like me) can show their (perhaps meagre) financial support.

**Comment by lukechu10** [comment]
Thanks for your interest in this project! I'm a bit hesitant to accept donations (at this moment) because it puts pressure on me to work on this project even when I'm busy with other things.

Code contributions will also be helpful and always be very welcome. Another thing that would help a lot would be to build some ecosystem crates. We already have some utilities in [`sycamore-hooks`](https://github.com/lukechu10/sycamore-hooks) which people could contribute to. Other bigger missing pieces such as animation, transitions, virtual lists, component libraries, form handling etc... would probably belong in their own crates however.

Lastly, other non-code resources such as blog posts would also be very helpful to promote the project and give valuable feedback on how to improve in the future.

Sorry for dumping everything here but it's just a few ideas for people who want to help out with sycamore. I am very grateful for the community around this project. Sycamore has gone beyond my wildest dreams: we have 1.2k stars on github and 300+ members on our discord server and still counting!

**Reply by davidatsurge** [reply]
This provides a lot of clarity, thanks!

I've joined the Discord and am glad to see a lot of activity. I hope to contribute along the lines of blogging and/or ecosystem crates as I keep using sycamore.
